source/Aconcagua-Quantitative-Analysis/DegreeCelsius.class.st

```diff
@@ -0,0 +1,80 @@
+"
+The degree Celsius is the unit of temperature on the Celsius temperature scale
+"
+Class {
+	#name : 'DegreeCelsius',
+	#superclass : 'UnitOfMeasure',
+	#instVars : [
+		'kelvin'
+	],
+	#category : 'Aconcagua-Quantitative-Analysis',
+	#package : 'Aconcagua-Quantitative-Analysis'
+}
+
+{ #category : 'instance creation' }
+DegreeCelsius class >> basedOn: kelvin [
```

Comment:
Why must kelvin be provided from outside the object? It seems like the class could be used in the wrong way with nothing to prevent it. It might be better to provide something that can confirm that the received kelvin is a valid implementation, or instance of it.

Reply:
Just because depending on the namespace makes the code more complex to compile in GS64, and complicates the initialization code. In any case, users are not supposed to instantiate this class directly but use it from the corresponding units namespace.